lib/signup_page.dart

```diff
@@ -0,0 +1,160 @@
+//bool _success;
+String _userEmail;
+String _password;
+
+class SignupPage extends StatefulWidget {
+  @override
+  _SignupPageState createState() => _SignupPageState();
+}
+
+class _SignupPageState extends State<SignupPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Container(
+        color: Colors.white,
+        child: Center(
+          child: Column(
+            mainAxisSize: MainAxisSize.max,
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              _form(context),
+              _back(context),
+              //_signUpButton(context),
+              //_signInButton(context),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+void signIn(_formKey) {}
+
+Widget _form(BuildContext context) {
+  final RegExp _emailValid = RegExp(
+      r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]+");
+
+  final RegExp _passwordValid =
+      RegExp(r'^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#\$&*~]).{8,}$');
+  RegExp regExp;
+  return Form(
+      key: _formKey,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: <Widget>[
+          TextFormField(
+            keyboardType: TextInputType.emailAddress,
+            decoration: const InputDecoration(
+              hintText: 'Email',
+            ),
+            validator: (value) {
+              if (value.isEmpty) {
+                return 'Email needed';
+              } else if (!_emailValid.hasMatch(value)) {
+                return 'Invalid Input';
+              }
+              _userEmail = value;
+              print('username good');
+              return null;
+            },
+            //controller: _emailController,
+            textAlign: TextAlign.center,
+          ),
+          TextFormField(
+            obscureText: true,
+            decoration: const InputDecoration(
+              hintText: 'Password',
+            ),
+            validator: (value) {
+              if (value.isEmpty) {
+                return 'Password Needed';
+              } else if (_passwordValid.hasMatch(value)) {
+                return 'something';
+              }
+              _password = value;
+              print('pass good');
+              return null;
+            },
+            textAlign: TextAlign.center,
+          ),
+          TextFormField(
+            obscureText: true,
+            decoration: const InputDecoration(
+              hintText: 'Confirm Password',
+            ),
+            validator: (value) {
+              if (value.isEmpty) {
+                return 'Retype Password';
+              } else if (_passwordValid.hasMatch(value)) {
+                return 'Password does not match';
+              }
+              _password = value;
+              print('Confirm pass good');
+              return null;
+            },
+            textAlign: TextAlign.center,
+          ),
```

Comment:
This validator is not working properly. It doesn't correctly check if the passwords don't match